exercises/react/example.cr

```diff
@@ -0,0 +1,71 @@
+alias CellValue = Int32
+
+private class Cell
+  @dependents = [] of ComputeCell
+
+  def initialize(initial : CellValue)
+    @value = initial
+  end
+
+  getter :value
+
+  protected getter :dependents
+  protected def propagate; _propagate; end
```

Comment:
Yeah.

```crystal
class Thing
  def initialize(@friends : Array(Thing) = [] of Thing)
  end

  protected def hi
    puts "hi from Thing"
  end
end

class SomeThing < Thing
  def hi_friends
    @friends.each(&.hi)
  end
end

class GoodThing < Thing
  protected def hi
    puts "hi from Good Thing"
  end
end

# The below line is a compile error! `protected method 'hi' called for GoodThing`
SomeThing.new([Thing.new, Thing.new]).hi_friends
SomeThing.new([Thing.new, GoodThing.new]).hi_friends

# REQUIREMENT: The below line MUST be a compile error.
# GoodThing.new.hi
# So it does not do to simply make `GoodThing#hi` public.
```

Interesting. And the minimal change to make it work is to make both `Thing` and `GoodThing` private. It does not do to make just one of them private.

Well, I am learning a lot today.